src/client/datascience/types.ts

```diff
@@ -233,14 +227,18 @@ export interface IJupyterPasswordConnect {
 }
 
 export const IJupyterSession = Symbol('IJupyterSession');
+/**
+ * Closely represents Jupyter Labs Kernel.IKernelConnection.
+ */
 export interface IJupyterSession extends IAsyncDisposable {
-    onSessionStatusChanged: Event<ServerStatus>;
-    onIOPubMessage: Event<KernelMessage.IIOPubMessage>;
-    readonly status: ServerStatus;
-    readonly workingDirectory: string;
+    readonly disposed: boolean;
+    readonly status: KernelMessage.Status;
     readonly kernelSocket: Observable<KernelSocketInformation | undefined>;
-    restart(timeout: number): Promise<void>;
-    interrupt(timeout: number): Promise<void>;
+    onSessionStatusChanged: Event<KernelMessage.Status>;
+    onDidDispose: Event<void>;
+    onIOPubMessage: Event<KernelMessage.IIOPubMessage>;
+    interrupt(): Promise<void>;
+    restart(): Promise<void>;
```

Comment:
This is very similar to Jupyters Kernel.IKernelConnection, which is what we want to get at eventually.